### PR TITLE
Setup CI, ESLint

### PR DESCRIPTION
**Pull request by hstove**
Note: I've put everything into smaller commits to help with reviewing, but we should rebase this into 1 commit before we merge.

- Sets up ESLint
  - Uses `@blockstack/eslint-config`, which I just created
  - Fixes all issues
- Sets up Github Actions
  - Adds CommitLint, tests, typecheck, linting

### CODE REVIEW

File: src/authorization.ts [hunk 1]
```diff
@@ -1,31 +1,14 @@
...
-
-import {
-  StacksPublicKey,
-  StacksPrivateKey
-} from './keys';
-
-import {
-  StacksMessage
-} from './message'
+import { AuthType, AddressHashMode, PubKeyEncoding, RECOVERABLE_ECDSA_SIG_LENGTH_BYTES } from './constants';
```
Comment (yknl) [inline src/authorization.ts]: I would prefer if we can keep a line length maximum. For me it really helps with readability. Is there any reason not to?

Comment (kyranjamie) [reply src/authorization.ts]: This is a Prettier rule we have set at 120.

We upped it from the default, 80, as with TypeScript it really isn't enough. With hindsight, I'm now of the opinion 120 is too long, and ~100 would be optimal.

Comment (yknl) [reply src/authorization.ts]: 100 is good enough for me

Comment (hstove) [reply src/authorization.ts]: Thanks @yknl , good suggestion! I've updated the local config for this project and fixed any resulting long lines. I would agree that the 100 char version is better.

Comment (kyranjamie) [reply src/authorization.ts]: https://github.com/blockstack/prettier-config/pull/1

**Pull request merged**